### PR TITLE
feat: allow using initials as `Avatar` placeholder

### LINKED ISSUE #357: Missing avatar with placeholder initials
**Describe the solution you'd like**

Flowbite React's Avatar component does not provide an avatar that has placeholder initials, unlike flowbite. E.g. https://flowbite.com/docs/components/avatar/ - see "Placeholder Initials".
The solution could be: instead of passing an image URL, Avatar could have another prop called "placeholderInitials" or similar where the developer passes the initials as a string.
Internally:
- If "img" props was provided, display the avatar with an image.
- If "placeholderIntitials" was provided, display the the initials only.

An suggestion: both should not be filled. If they are, avatar with an image should be display or just throw an error.

### PR DESCRIPTION
**Pull request by iamargentum**
Refs: #357

## Description

This PR adds a new prop, `placeholderInitials`, to the `Avatar` component. String passed in this prop will be displayed as `Avatar`'s placeholder, if image is not provided.

Fixes #357 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change contains documentation update

## Breaking changes

No breaking changes

## How Has This Been Tested?

Just checked visually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

[Screencast from 10-10-22 01:57:39 AM IST.webm](https://user-images.githubusercontent.com/51338731/194778097-ea0ceee2-d2be-4763-8267-54231c806110.webm)

### CODE REVIEW

File: src/lib/components/Avatar/index.tsx [hunk 1]
```diff
@@ -52,6 +54,10 @@ const AvatarComponent: FC<AvatarProps> = ({
             data-testid="flowbite-avatar-img"
             src={img}
           />
+        ) : placeholderInitials ? (
+          <div className="inline-flex overflow-hidden relative justify-center items-center w-10 h-10 bg-gray-100 rounded-full dark:bg-gray-600">
+            <span className="font-medium text-gray-600 dark:text-gray-300">{placeholderInitials}</span>
```
Comment (rluders) [inline src/lib/components/Avatar/index.tsx:59]: Same thing here.

**Review by rluders — CHANGES_REQUESTED** [review]

File: src/lib/components/Avatar/index.tsx (hunk 1 above)
Comment (iamargentum) [reply src/lib/components/Avatar/index.tsx]: @rluders addressed this in the latest commit

**Review by rluders — APPROVED** [review]
Thanks.

**Comment by rluders** [comment]
@iamargentum I guess that now the CI is just saying that we need to put some unit test to cover the new logic. :)

**Comment by iamargentum** [comment]
okay!
I've never really written a test though
i'll try to replicate what existing tests for the new logic

**Pull request merged**